polyhost/gui/layout_dialog/kb_layout_dialog.py

```
@@ -86,6 +89,22 @@ def __init__(self, keeb: PolyKybd, settings: DeviceSettings, parent=None):
+        """Run a device read through the worker. If the worker is suspended
+        (e.g. during a firmware flash) run_sync raises RuntimeError — surface the
+        existing not-connected message box and return ``default`` instead of
+        crashing."""
+        try:
+            return self.worker.run_sync(name, fn, timeout=5)
+        except RuntimeError:
+            QMessageBox.warning(
+                None, "Not Connected",
+                "PolyKybd is not connected. Please connect the keyboard and try again.")
+            return default
+        except Exception as exc:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)